src/service-override/files.ts

```diff
@@ -194,8 +194,9 @@ class RegisteredFileSystemProvider extends Disposable implements IFileSystemProv
 
   public async readdir (resource: URI): Promise<[string, FileType][]> {
     const includedPaths = Array.from(this.files.keys())
-      .map(uri => extUri.relativePath(resource, URI.parse(uri))!)
-      .filter(path => !path.startsWith('..'))
+      .map(uri => URI.parse(uri))
+      .filter(uri => uri.path.startsWith(resource.path))
```

Comment:
I think this can be a problematic check. For example, if the directory is `/a`, then all files that start with `/a` will be included. I think the check should be something along the lines of:

```
.filter(uri => uri.path.startsWith(resource.path + '/'))
```

I've been bitten by a check like this before, maybe it does not apply here.